icu4j/main/classes/core/src/com/ibm/icu/util/CECalendar.java

```diff
@@ -86,7 +86,10 @@ protected CECalendar(Locale aLocale) {
      * @param locale The locale for the new calendar.
      */
     protected CECalendar(ULocale locale) {
-        this(TimeZone.getDefault(), locale);
+        this((locale.getKeywordValue("timezone") == null)
```

Comment:
Why didn't you make the `forLocaleOrDefault` change in Java, too?

Reply:
ok added.